packages/streamline/src/components/inputs/input-text/components/input-text-primitive.tsx

```diff
@@ -0,0 +1,188 @@
+import { ColorTheme, useStreamlineTheme } from '../../../../theme';
+import React from 'react';
+import {
+  TextInput as NativeTextInput,
+  TextInputProps,
+  View,
+} from 'react-native';
+import { IconAdornment } from './input-text-icon';
+import { InputTextLabel } from './input-text-label';
+import { InputTextOutline, OUTLINE_WIDTH } from './input-textoutline';
+
+import { TextInputPrimitiveProps } from './types';
+
+const TEXT_INPUT_HEIGHT_WITHOUT_LABEL = 48;
+const DEFAULTLINEHEIGHT = 18;
+
+export const InputTextPrimitive = ({
+  isSearchInput,
+  isDisabled,
+  isEditable,
+  isError,
+  description,
+  multiline,
+  numberOfLines = 1,
+  isFocused,
+  label,
+  placeholder,
+  secureTextEntry,
+  value,
+  onFocus,
+  onBlur,
+  onChangeText,
+  right,
+  left,
+  forceFocus,
+  parentState,
+  innerRef,
+  pointerEvents,
+  render = (props: TextInputProps) => <NativeTextInput {...props} />,
+  maxWidth,
+  width,
+  height,
+  noPadding,
+  ...rest
+}: TextInputPrimitiveProps) => {
+  const theme = useStreamlineTheme();
+  const gutterStyle = { marginLeft: left ? theme.spacing.xs : 0 };
+  const containerStyle = !label &&
+    !isSearchInput && { paddingTop: theme.spacing.xxs };
+  const inputPadding = { paddingVertical: theme.spacing.xs + OUTLINE_WIDTH };
+  const inputContainer = (
+    <Box flexGrow={1} flexBasis={0}>
+      {render?.({
+        ...rest,
+        maxFontSizeMultiplier: 1,
+        ref: innerRef,
+        selectionColor: theme.colors.BLUKO_1000,
+        placeholder: placeholder,
+        placeholderTextColor: theme.colors.GREY_300,
+        secureTextEntry: secureTextEntry,
+        editable: !isDisabled && isEditable,
+        multiline: multiline,
+        numberOfLines: numberOfLines,
+        value: value,
+        onFocus: onFocus,
+        onBlur: onBlur,
+        onChangeText: onChangeText,
+        style: [
+          {
+            flexGrow: 1,
+            color: theme.colors.GREY_1000,
+            fontFamily: theme.textVariants.body.fontFamily,
+            fontSize: theme.textVariants.body.fontSize,
+            minHeight: numberOfLines * DEFAULTLINEHEIGHT,
+          },
+          isSearchInput
+            ? {
+                marginLeft: theme.spacing.xxs,
+              }
+            : inputPadding,
+          isDisabled && { color: theme.colors.GREY_500 },
+          isError && isDisabled && { color: theme.colors.GREY_500 },
+          // fix for dot secure text size who are too big with circular font
+          secureTextEntry && value ? { fontFamily: 'System' } : {},
+          maxWidth ? { maxWidth } : undefined,
+          rest.style,
+        ],
+      })}
+    </Box>
+  );
+  return (
+    <View style={containerStyle} pointerEvents={pointerEvents}>
+      {!!label && (
+        <Box paddingBottom="xs">
+          <InputTextLabel
+            label={label}
+            active={isFocused}
+            disabled={isDisabled}
+          />
+        </Box>
+      )}
```

Comment:
Sometimes `&&` is used and other times `? : null`, the usage should be consistent